storage/merkle/tree.go

```diff
@@ -218,78 +264,79 @@ GetLoop:
 
 		// if we have a leaf, we found the key, return value and true
 		case *leaf:
-
 			return n.val, true
 
 		// if we have a nil node, key doesn't exist, return nil and false
 		case nil:
-
 			return nil, false
 		}
 	}
 }
 
-// Del will remove the value associated with the given key from the patricia
-// merkle trie. It will return true if they key was found and false otherwise.
+// Del removes the value associated with the given key from the patricia
+// merkle trie. It returns true if they key was found and false otherwise.
 // Internally, any parent nodes between the leaf up to the closest shared path
 // will be deleted or merged, which keeps the trie deterministic regardless of
 // insertion and deletion orders.
 func (t *Tree) Del(key []byte) bool {
+	if t.root == nil || t.keyLength != len(key) {
```

Comment:
and I guess it's the same here for `unsafeDel`, but it's better to double check :)
```suggestion
	if t.keyLength != len(key) {
```

Reply:
👍